lib/audited/auditor.rb

```diff
@@ -196,11 +197,35 @@ def revision_with(attributes)
 
       def audited_changes
         all_changes = respond_to?(:changes_to_save) ? changes_to_save : changes
-        if audited_options[:only].present?
-          all_changes.slice(*self.class.audited_columns)
-        else
-          all_changes.except(*self.class.non_audited_columns)
+        filtered_changes = \
+          if audited_options[:only].present?
+            all_changes.slice(*self.class.audited_columns)
+          else
+            all_changes.except(*self.class.non_audited_columns)
+          end
+
+        filtered_changes = normalize_enum_changes(filtered_changes)
+        filtered_changes.to_hash
+      end
+
+      def normalize_enum_changes(changes)
+        self.class.defined_enums.each do |name, values|
+          if changes.has_key?(name)
+            changes[name] = \
+              if changes[name].is_a?(Array)
+                changes[name].map { |v| values[v] }
+              elsif rails_below?('5.0')
+                changes[name]
```

Comment:
Why for rails<5 this is a noop but only if the change isn't an array?

Reply:
That is due to the fact, that rails stores enum changes differently for rails < 5 and for rails >= 5.